tasks.py

```diff
@@ -782,14 +785,74 @@ def show_automerges(ctx: Context):
         print("\tNo PRs set to automerge")
 
 
-MARKER_REQ_MAPPING: Final[Mapping[str, tuple[str, ...]]] = {
-    "athena": ("reqs/requirements-dev-athena.txt",),
-    "cloud": ("reqs/requirements-dev-cloud.txt",),
-    "pyarrow": ("reqs/requirements-dev-arrow.txt",),
-    "external_sqldialect": ("reqs/requirements-dev-sqlalchemy.txt",),
+class TestDependencies(NamedTuple):
+    requirement_files: tuple[str, ...]
+    services: tuple[str, ...] = tuple()
+    setup_funcs: tuple[Callable[[Context], Result | None], ...] = tuple()
+    exta_pytest_args: tuple[  # TODO: remove this once remove the custom flagging system
+        str, ...
+    ] = tuple()
+    # NOTE: We probably don't need both `services` and `setup_funcs`
+    # but it's not clear to me which one is better
+    # `services` is simpler because it's just a list of services to up with docker-compose
+    # but `setup_funcs` is more flexible because it allows us to do more complex setup
+    # It could do the same docker-compose up or anything else.
+    # setup_funcs is a tuple of functions that take a an `invoke.Context` and return an invoke `Result` or `None`
+    # The invoke context can be used to run CLI subcommands or can be ignored.
+    # More arguments could be added to this `Callable` interface if needed.
+
+
+MARKER_DEPENDENDENCY_MAP: Final[Mapping[str, TestDependencies]] = {
+    "athena": TestDependencies(("reqs/requirements-dev-athena.txt",)),
+    "cloud": TestDependencies(
+        ("reqs/requirements-dev-cloud.txt",), exta_pytest_args=("--cloud",)
+    ),
+    "external_sqldialect": TestDependencies(("reqs/requirements-dev-sqlalchemy.txt",)),
+    "pyarrow": TestDependencies(("reqs/requirements-dev-arrow.txt",)),
+    "postgres": TestDependencies(
+        ("reqs/requirements-dev-postgresql.txt",),
+        # example of using a simple service lookup and a setup function to do the same thing
+        # we can remove one of these methods if one seems obviously better
+        services=("postgres",),
+        setup_funcs=(
+            lambda ctx: print("Dummy setup function"),
+            # lambda ctx: ctx.run("docker-compose up -d postgres"),
+        ),
+        exta_pytest_args=("--postgresql",),
+    ),
```

Comment:
I believe I've left this PR in a state that only requires changes to the new `docker-compose.yaml` file and this mapping.

Hopefully, the examples make how to add new services and/or `setup_funcs` clear.

No changes should need to be make to `invoke task` logic or the github actions CI (apart from adding new markers to the matrix).

As of the time of this posting the postgres test fails because the postgres test clients aren't able to connect to the service (I'm sure @nicgrayson or @billdirks  could resolve that without too much trouble). 
However, if we just want to get this merged all that is required is to comment out the `postgres` entry for the test matrix.